Werkgroep API strategie/principes.md

```diff
@@ -0,0 +1,216 @@
+
+API's die vrije-tekst zoeken ondersteunen doen dit middels de query-parameter zoek.
+
+## API-37: API's die vrije-tekst zoeken ondersteunen kunnen overweg met twee soorten wildcard karakters: `*` en `?`
+
+API's die vrije-tekst zoeken ondersteunen kunnen overweg met twee soorten wildcard karakters:
+
+- `*` Komt overeen met nul of meer (niet-spatie) karakters
+- `?` Komt precies overeen met één (niet-spatie) karakter
+
+## API-38: GEO API's ontvangen en versturen GeoJSON
+
+Voor GEO API's wordt de standaard GeoJSON gebruikt.
+
+## API-39: GeoJSON is onderdeel van de embedded resource in de JSON response
+
+GeoJSON wordt in een JSON response (application/json) geplaatst waarbij geo attributen als GeoJSON-compatible object in de resource ge-embed zijn.
+
+## API-40: Voor GEO queries is een POST end-point beschikbaar
+
+Geometrische queries worden in een POST naar een apart end-point verzonden.
+
+## API-41: POST end-points zijn uitbreidbaar voor gecombineerde vragen
+
+De geometrie is uitbreidbaar met andere properties voor gecombineerde queries.
+
+## API-42: Resultaten van een globale geometrische zoekvraag worden in de relevante geometrische context geplaatst
+
+In het geval van een globale zoekvraag …/api/zoek/v1 is het noodzakelijk om de resultaten in een relevante geometrische context te plaatsen.
+
+## API-43: Het voorkeur-coördinatenstelsels (CRS) is ETRS89, maar het CRS wordt niet impliciet geselecteerd
+
+Algemeen gebruik van het Europese ETRS89 coördinatenstelsel (CRS) heeft sterk de voorkeur, maar dit wordt niet door API's afgedwongen als de "default". Derhalve moet het te gebruiken CRS in elke aanroep expliciet worden opgegeven.
+
+## API-44: Het coördinatenstelsel (CRS) van de vraag en het antwoord wordt in de header meegestuurd
+
+Het gekozen coördinatenstelsel (CRS) voor zowel de vraag als het antwoord worden meegestuurd als onderdeel van de request-header en de response-header. Bij het ontbreken van deze headers wordt 412 Precondition Failed teruggegeven. Hiermee wordt voorkomen dat per abuis met het verkeerde CRS wordt gewerkt.
+
+## API-45: Het gewenste coördinatenstelsel wordt op basis van content negotiation overeengekomen
+
+Het gewenste CRS voor de geometrie in het antwoord (response body) wordt aangeduid met een accept header. Als het gewenste CRS niet geleverd kan worden zal er een 406 Not Acceptable worden teruggegeven.
+
+## API-46: Paginering wordt gerealiseerd op basis van JSON+HAL
+
+Aan de representatie worden twee gereserveerde velden (gedefinieerd door RFC5988) "_links" (verplicht) en _embedded (optioneel) toegevoegd. Daarnaast wordt paginerings-metadata als HTTP-headers meegegeven.
+
+## API-47: Waar relevant wordt caching toegepast
+
+Voor caching wordt gebruikt van de HTTP standaard caching mechanismes door het toevoegen van een aantal extra uitgaande HTTP-headers (ETag of Last-Modified) en functionaliteit om te bepalen of een aantal specifieke inkomende HTTP-headers (Is-None_Match of Is-Modified-Since).
```

Comment:
Those are not only for caching, but for [HTTP RFC: Conditional Requests]( https://tools.ietf.org/html/rfc7232).

For Caching you could check [HTTP RFC: Caching](https://tools.ietf.org/html/rfc7234) and this nice piece from mozilla: https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching